### PR TITLE
Shared Choose Branch Dialog

### PR DESCRIPTION
**Pull request by tidy-dev**
## Description

A choose branch dialog for merge, squash merge, and rebase to share. The `BaseChooseBranchDialog` has the shared logic of the Merge and Rebase choose branch dialog and adds in the new dropdown select button. The `MergeChooseBranchDialog` extends this and holds the logic specific to merge. (Later will add one for Rebase). The squash merge will just use the Merge as it won't be much different.

## Release notes

Notes: no-notes

### CODE REVIEW

File: app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+} from '../../../lib/git'
+import { promiseWithMinimumTimeout } from '../../../lib/promise'
+import { Branch } from '../../../models/branch'
+import { ComputedAction } from '../../../models/computed-action'
+import { MergeTreeResult } from '../../../models/merge'
+import { MultiCommitOperationKind } from '../../../models/multi-commit-operation'
+import { ActionStatusIcon } from '../../lib/action-status-icon'
+import { BaseChooseBranchDialog } from './base-choose-branch-dialog'
+
+export abstract class MergeChooseBranchDialog extends BaseChooseBranchDialog {
```
Comment (sergiou87) [inline app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx]: This one shouldn't be abstract, right?
```suggestion
export class MergeChooseBranchDialog extends BaseChooseBranchDialog {
```

Interestingly, TypeScript doesn't complain if you instance this class using JSX :til:

**Review by sergiou87 — CHANGES_REQUESTED** [review]
LGTM! Only one nit 😄

File: app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx (hunk 1 above)
Comment (tidy-dev) [reply app/src/ui/multi-commit-operation/choose-branch/merge-choose-branch-dialog.tsx]: You are correct! Copy/Pasta mistake. Thanks!

**Review by sergiou87 — APPROVED** [review]
:shipit:

**Pull request merged**